### PR TITLE
Implement column width/spacing

### PR DESCRIPTION
**Pull request by smoogipoo**
Resolves the part of https://github.com/ppy/osu/issues/8530 where the notes have different sizes.

### CODE REVIEW

File: osu.Game.Rulesets.Mania/UI/ManiaStage.cs [hunk 1]
```diff
@@ -150,6 +150,41 @@ public ManiaStage(int firstColumnIndex, StageDefinition definition, ref ManiaAct
...
+
+                    col.Margin = new MarginPadding { Left = spacing };
+                }
+
+                float? width = currentSkin.GetConfig<LegacyManiaSkinConfigurationLookup, float>(
+                                              new LegacyManiaSkinConfigurationLookup(Columns.Count, LegacyManiaSkinConfigurationLookups.ColumnWidth, col.Index))
+                                          ?.Value;
+
+                if (width == null)
+                    col.Width = col.IsSpecial ? Column.SPECIAL_COLUMN_WIDTH : Column.COLUMN_WIDTH;
```
Comment (mcendu) [inline osu.Game.Rulesets.Mania/UI/ManiaStage.cs:182]: Most skins assume that the column width is 30 instead of the lazer values. Might be another PR imo.

Comment (smoogipoo) [reply osu.Game.Rulesets.Mania/UI/ManiaStage.cs:182]: Check the line below this one, where I'm using the skin width.

File: osu.Game.Rulesets.Mania/UI/ManiaStage.cs [hunk 2]
```diff
@@ -150,6 +150,41 @@ public ManiaStage(int firstColumnIndex, StageDefinition definition, ref ManiaAct
...
+            skin.SourceChanged += onSkinChanged;
+
+            onSkinChanged();
+        }
+
+        private void onSkinChanged()
+        {
+            foreach (var col in columnFlow)
+            {
+                if (col.Index > 0)
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/ManiaStage.cs:168]: what does index 0 mean here?

Comment (smoogipoo) [reply osu.Game.Rulesets.Mania/UI/ManiaStage.cs:168]: The first column. It doesn't have spacing.

Comment (peppy) [reply osu.Game.Rulesets.Mania/UI/ManiaStage.cs:168]: aha, missed that

File: osu.Game.Rulesets.Mania/UI/ManiaStage.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Rulesets.Mania/UI/ManiaStage.cs:182]: this is only ever hit by the default skin, correct?

i think we may want to re-think this lookup process. at very least the configuration lookup type should be renamed from "legacy" if we are using it in a mainstream class (`ManiaStage`)

Comment (smoogipoo) [reply osu.Game.Rulesets.Mania/UI/ManiaStage.cs:182]: Yeah there's some configs that are 100% legacy, and others that probably aren't going to be, such as column width and spacing.

I didn't split it out because I'm not sure I want to implement 3 ways of retrieving mania skin stuff though: `ManiaSkinComponent`, `LegacyManiaSkinConfigurationLookup`, `ManiaSkinConfigurationLookup`, all before we have a system for dealing with drawable configs.

**Review by peppy — APPROVED** [review]
